### PR TITLE
Add owner attribute to BranchInfo

### PR DESCRIPTION
**Pull request by EyalDelarea**
- [x] All [tests](https://github.com/jfrog/froggit-go/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `go fmt ./...` for formatting the code before submitting the pull request.
- [x] This feature is included on all supported VCS providers - GitHub, Bitbucket cloud, Bitbucket server, and GitLab.

---

Add owner attribute to branch info, when getting pull request info it is important to know both of the owners, as they can differ when the pull request is from a forked repository.

### CODE REVIEW

File: vcsclient/azurerepos.go [hunk 1]
```diff
@@ -568,20 +568,44 @@ func parsePullRequestDetails(pullRequest git.GitPullRequest, repository string)
...
+func extractOwnerFromForkedRepoUrl(forkedGit *git.GitForkRef) string {
+	if forkedGit == nil || forkedGit.Repository == nil || forkedGit.Repository.Url == nil {
+		return ""
+	}
+	url := *forkedGit.Repository.Url
+	owner := strings.Split(strings.TrimPrefix(url, "https://dev.azure.com/"), "/")[0]
+	// Failed to properly extract owner name
+	if strings.Contains(owner, "http") {
+		return ""
+	}
```
Comment (omerzi) [inline vcsclient/azurerepos.go:605]: If we fail why the owner string will contain the "http"? Can you elaborate on this scenario?

File: vcsclient/bitbucketcloud.go [hunk 2]
```diff
@@ -782,3 +788,11 @@ func getBitbucketCloudRepositoryVisibility(repo *bitbucket.Repository) Repositor
 	}
 	return Public
 }
+
+func splitWorkSpaceAndOwner(name string) (string, string) {
```
Comment (omerzi) [inline vcsclient/bitbucketcloud.go:792]: Let's add a description of this function. What is the expected format and the result

File: vcsclient/bitbucketserver.go [hunk 3]
```diff
@@ -412,10 +412,12 @@ func (client *BitbucketServerClient) GetPullRequestByID(ctx context.Context, own
 	if err != nil {
 		return
 	}
+	// Could differ from a forked repository
+	sourceOwner := pullRequest.FromRef.Repository.Project.Key
```
Comment (omerzi) [inline vcsclient/bitbucketserver.go:416]: Project could be nil, let's add a nil check

File: vcsclient/github.go [hunk 4]
```diff
@@ -342,10 +342,12 @@ func (client *GitHubClient) GetPullRequestByID(ctx context.Context, owner, repos
 		Source: BranchInfo{
 			Name:       sourceBranch,
 			Repository: *pullRequest.Head.Repo.Name,
+			Owner:      *pullRequest.Head.Repo.Owner.Login,
```
Comment (omerzi) [inline vcsclient/github.go:345]: We should check for nil values for all of this function... You can use `DefaultIfNotNil` func, under `vcsutils/utils.go`
Let's apply this function for all over

File: vcsutils/utils.go [hunk 5]
```diff
@@ -327,3 +332,17 @@ func MapPullRequestState(state *PullRequestState) *string {
 	}
 	return &stateStringValue
 }
+
+func GetDefaultApiEndpoint(provider VcsProvider) string {
```
Comment (omerzi) [inline vcsutils/utils.go:336]: Unused function

**Review by omerzi — CHANGES_REQUESTED** [review]

File: vcsclient/azurerepos.go (hunk 1 above)
Comment (EyalDelarea) [reply vcsclient/azurerepos.go:605]: You're right, it wasn't clear. i've changed it 👍

File: vcsutils/utils.go (hunk 5 above)
Comment (EyalDelarea) [reply vcsutils/utils.go:336]: It is consumed by Frogbot.

**Pull request merged**